client/src/store/slices/mentor/mentorSlice.ts

```diff
@@ -0,0 +1,151 @@
+/*
+This software is Copyright ©️ 2020 The University of Southern California. All Rights Reserved. 
+Permission to use, copy, modify, and distribute this software and its documentation for educational, research and non-profit purposes, without fee, and without a written agreement is hereby granted, provided that the above copyright notice and subject to the full license file found in the root of this software deliverable. Permission to make commercial use of this software may be obtained by contacting:  USC Stevens Center for Innovation University of Southern California 1150 S. Olive Street, Suite 2300, Los Angeles, CA 90115, USA Email: accounting@stevens.usc.edu
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";
+import * as api from "api";
+import { LoadingError } from "hooks/graphql/loading-reducer";
+import { Mentor } from "types";
+
+/** Store */
+
+export enum MentorStatus {
+  NONE = 0,
+  LOADING = 1,
+  SAVING = 2,
+  SUCCEEDED = 3,
+  FAILED = 4,
+}
+
+export interface MentorState {
+  data: Mentor | undefined;
+  editedData: Mentor | undefined;
+  isEdited: boolean;
+  mentorStatus: MentorStatus;
+  error: LoadingError | undefined;
+}
+
+const initialState: MentorState = {
+  data: undefined,
+  editedData: undefined,
+  isEdited: false,
+  mentorStatus: MentorStatus.NONE,
+  error: undefined,
+};
+
+/** Actions */
+
+export const loadMentor = createAsyncThunk(
+  "mentor/loadMentor",
+  async (accessToken: string) => {
+    return api.fetchMentor(accessToken);
+  }
+);
+
+export const saveMentor = createAsyncThunk(
+  "mentor/saveMentor",
+  async (
+    headers: { accessToken: string; editedData: Mentor },
+    { rejectWithValue }
+  ) => {
+    try {
+      return api.updateMentorDetails(headers.editedData, headers.accessToken);
+    } catch (err) {
```

Comment:
```typescript
// needs await because you're looking for the response in the catch
return await api.updateMentorDetails(headers.editedData, headers.accessToken);
```